src/components/I18nProvider.js

```diff
@@ -0,0 +1,25 @@
+import { Component } from 'preact';
+
+import getTranslations from '../utils/i18n';
+
+export default class I18nProvider extends Component {
+    constructor(props) {
+        super(props);
+
+        const { language } = this.props;
+
+        this.state = { i18n: getTranslations(language) };
```

Comment:
do we need to keep it in the state? I propose to have ```this.i18n = getTranslations(language)```

Reply:
maybe we don't, I'll try to move that invocation directly to `getChildContext` instead of having such field